### PR TITLE
[PF-749] fix tag publish

### PR DESCRIPTION
**Pull request by ddietterich**
Skip the test workflow and the tag-publish workflow if the push to dev commit message is `"^bump .*"`. This will save a complete, redundant test pass and tag computation when merging code.

I added a composite action, `bump-skip`, that performs the check and sets an output. Then each following step includes an `if:` that checks against that output.

There is no "exit at this point" logic in github actions, so the `if:` approach is the recommended solution. An alternative is to use sequential jobs and put the `if:` at the job level. However, that would mean two separate checkouts, so the step level seemed like the more straightforward approach.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -69,16 +77,19 @@ jobs:
...
       run: psql -h 127.0.0.1 -U postgres -f ./service/local-dev/local-postgres-init.sql
 
     - name: Set up AdoptOpenJDK 11
+      if: steps.skiptest.outputs.is-bump == 'no'
       uses: joschi/setup-jdk@v2
       with:
         java-version: 11
 
     - name: Cache Gradle packages
+      if: steps.skiptest.outputs.is-bump == 'no'
```
Comment (jaycarlton) [inline .github/workflows/test.yml:92]: It's amazing they don't support yaml anchors for this kind of thing https://github.community/t/support-for-yaml-anchors/16128/10

**Review by zloery — APPROVED** [review]

File: .github/actions/bump-skip/action.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  using: "composite"
+  steps:
+    - name: Bump test
+      id: bumptest
+      run: |
+        log=$(git log --pretty='%B')
+        echo "log=$log"
+        pattern="^bump .*"
+        IS_BUMP=no
+        if [[ "${{ inputs.event-name }}" == "push" && "$log" =~ $pattern ]]; then
```
Comment (jaycarlton) [inline .github/actions/bump-skip/action.yml:23]: do you need quotes around `$pattern`?

File: .github/workflows/tag-publish.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Set part of semantic version to bump
-      id: semver
+      id: controls
```
Comment (jaycarlton) [inline .github/workflows/tag-publish.yml:49]: I don't get this name.

File: .github/actions/bump-skip/action.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          IS_BUMP=yes
+        fi
+        echo "IS_BUMP=$IS_BUMP"
```
Comment (jaycarlton) [inline .github/actions/bump-skip/action.yml:26]: Why does it take two echo statements?

**Review by jaycarlton — APPROVED** [review]
This should free up a bunch of cycles. Can it be shared with other repos/the community?

File: .github/actions/bump-skip/action.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    value: ${{ steps.bumptest.outputs.is-bump }}
+runs:
+  using: "composite"
+  steps:
+    - name: Bump test
+      id: bumptest
+      run: |
+        log=$(git log --pretty='%B')
+        echo "log=$log"
+        pattern="^bump .*"
```
Comment (jaycarlton) [inline .github/actions/bump-skip/action.yml:21]: We could generalize the pattern to something like `[skip_tests]` and allow it anywhere in the log line, but I guess that's a bit ugly in the log

File: .github/actions/bump-skip/action.yml (hunk 4 above)
Comment (ddietterich) [reply .github/actions/bump-skip/action.yml:26]: One echo is for the log so you know whether it is a bump. The other echo is magic that sets an output variable from the action.

File: .github/actions/bump-skip/action.yml (hunk 5 above)
Comment (ddietterich) [reply .github/actions/bump-skip/action.yml:21]: Agreed. I made another PR. It requires a change to the common bumper action and the a compensating change to our GHAs.

**Pull request merged**